src/data_type.hpp

```diff
@@ -365,11 +367,15 @@ class UserType : public DataType {
       return false;
     }
 
-    if (fields_.size() != user_type->fields_.size()) {
-      return false;
-    }
 
-    for (size_t i = 0; i < fields_.size(); ++i) {
+    // UDT's can be considered equal as long as the mutual first fields shared
+    // between them are equal. UDT's are append only as far as fields go, so a
+    // newer 'version' of the UDT data type after a schema change event should be
+    // treated as equivalent in this scenario, by simply looking at the first N
+    // mutual fields they should share.
+    size_t min_fields_size = std::min(fields_.size(), user_type->fields_.size());
```

Comment:
This seems reasonable to me. It does seem that after creating a type that a type can only be altered in the following ways:

* Rename field: `ALTER TYPE address RENAME zip TO zipcode AND street_name TO street` 
* Add a field: `ALTER TYPE address ADD country text`

It'd be nice to have some tests that verify this is a correct assumption.

Reply:
> It'd be nice to have some tests that verify this is a correct assumption.

I'm not entirely sure how we would test that? Are you suggesting we try to test that the database rejects invalid DDL? 

There is a third way, you can change the type of a field `ALTER field_name TYPE new_cql_datatype` - however, I think if you're doing this, you probably need to redeploy your application anyways since the types are changing.